projects/plugins/boost/app/modules/cache/Page_Cache.php

```diff
@@ -34,173 +36,24 @@ class Page_Cache implements Pluggable, Is_Always_On {
 
 	public function __construct() {
 		$this->settings = Boost_Cache_Settings::get_instance();
-		register_deactivation_hook( JETPACK_BOOST_PATH, array( $this, 'deactivate' ) );
-		register_uninstall_hook( JETPACK_BOOST_PATH, 'Page_Cache::uninstall' );
+		register_deactivation_hook( JETPACK_BOOST_PATH, array( Page_Cache_Setup::class, 'deactivate' ) );
+		register_uninstall_hook( JETPACK_BOOST_PATH, array( Page_Cache_Setup::class, 'uninstall' ) );
 	}
 
-	/*
-	 * Sets up the advanced-cache.php file and if that works, adds the WP_CACHE
-	 * define to wp-config.php
-	 * These are used by WordPress to load the caching system before most of
-	 * WordPress is loaded.
-	 */
-	public function setup() {
```

Comment:
Moving this all out to a special setup class responsible for setting up and tearing down was a nice idea 👍